### PR TITLE
Update TSP query parameters descriptions and schemas

### PR DESCRIPTION
**Pull request by PatrickTasse**
Signed-off-by: Patrick Tasse <patrick.tasse@gmail.com>

### CODE REVIEW

File: API.yaml [hunk 1]
```diff
@@ -715,16 +774,36 @@ paths:
...
+                    requested_items:
+                      type: array
+                      items:
+                        type: integer
+                    xValue:
+                      type: integer
+                    yValue:
+                      type: number
+                  required:
+                    - requested_times
```
Comment (PatrickTasse) [inline API.yaml:796]: Should be xValue

File: API.yaml [hunk 2]
```diff
@@ -769,14 +848,21 @@ paths:
...
         description: Query parameters to fetch the timegraph tree.
-          Usually contains these parameters, but not limited to
-          - startTime
-          - endTime
         required: true
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/Query'
+              type: object
```
Comment (bhufmann) [inline API.yaml:855]: should we add requested_times as well like in XY?

File: API.yaml [hunk 3]
```diff
@@ -930,14 +1057,34 @@ paths:
...
         description: Query parameters to fetch the tooltips.
-          Usually contains these parameters, but not limited to
-          - entries, array of all the entry (series) IDs
-          - times, array of all the times
         required: true
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/Query'
+              type: object
```
Comment (bhufmann) [inline API.yaml:1064]: Tooltips can be a GET. All the implementations are so far are GET. Let's align it to GET given the current deadline. If needed later on we can change it to POST.

File: API.yaml [hunk 4]
```diff
@@ -826,17 +912,44 @@ paths:
...
+              properties:
+                parameters:
+                  type: object
+                  properties:
+                    requested_times:
+                      type: array
+                      items:
+                        type: integer
+                    requested_items:
```
Comment (bhufmann) [inline API.yaml:928]: can we add a description (also for every parameter)?

File: API.yaml [hunk 5]
```diff
@@ -826,17 +912,44 @@ paths:
...
+                      type: array
+                      items:
+                        type: integer
```
Comment (bhufmann) [inline API.yaml:931]: can we add a description?

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: API.yaml [hunk 6]
```diff
@@ -659,16 +685,35 @@ paths:
...
+                  required:
+                    - requested_times
+                    - requested_items
```
Comment (bhufmann) [inline API.yaml:708]: requested_item should be also described in the description above

File: API.yaml [hunk 7]
```diff
@@ -769,14 +819,27 @@ paths:
           type: string
       requestBody:
         description: Query parameters to fetch the timegraph tree.
-          Usually contains these parameters, but not limited to
-          - startTime
-          - endTime
+          When 'requested_times' is absent the tree for the full range returned. When size is 2 it is the start and end of the requested range, otherwise it is the explicit array of requested sample times.
```
Comment (bhufmann) [inline API.yaml:822]: same comment as above

File: API.yaml [hunk 8]
```diff
@@ -826,17 +889,35 @@ paths:
           type: string
       requestBody:
         description: Query parameters to fetch the timegraph states.
-          Usually contains these parameters, but not limited to
-          - entries, array of all the entry (series) IDs
-          - nb, number of time stamps to sample
-          - startTime
-          - endTime
-          - labelRatio, ratio of state duration to label's string length to determine if the resolution is large enough to return the label
+         When 'requested_times' size is 2 it is the start and end of the requested range, otherwise it is the explicit array of requested sample times.
```
Comment (bhufmann) [inline API.yaml:892]: same comment as above

File: API.yaml [hunk 9]
```diff
@@ -879,14 +960,29 @@ paths:
           type: string
       requestBody:
         description: Query parameters to fetch the timegraph arrows.
-          Usually contains these parameters, but not limited to
-          - startTime
-          - endTime
+          When 'requested_times' size is 2 it is the start and end of the requested range, otherwise it is the explicit array of requested sample times.
```
Comment (bhufmann) [inline API.yaml:963]: same comment than above

File: API.yaml [hunk 10]
```diff
@@ -920,24 +1016,30 @@ paths:
...
+        in: query
+        description: The time for which to get the tooltip
         required: true
-        content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/Query'
+        schema:
+          type: integer
+      - name: entryId
```
Comment (bhufmann) [inline API.yaml:1033]: this one is required.

File: API.yaml [hunk 11]
```diff
@@ -920,24 +1016,30 @@ paths:
...
+        in: query
+        description: The entryId for which to get the tooltip
+        schema:
+          type: integer
+      - name: targetId
+        in: query
+        description: The arrow targetId for which to get the tooltip
```
Comment (bhufmann) [inline API.yaml:1040]: This one is optional. Only needed for arrows. Maybe add this comment.

File: API.yaml [hunk 12]
```diff
@@ -987,19 +1089,36 @@ paths:
...
+              type: object
+              properties:
+                parameters:
+                  type: object
+                  properties:
+                    requested_times:
+                      type: array
+                      items:
+                        type: integer
+                    requested_items:
```
Comment (bhufmann) [inline API.yaml:1107]: describe in description

File: API.yaml [hunk 13]
```diff
@@ -1091,16 +1218,41 @@ paths:
         schema:
           type: string
       requestBody:
-        description: Query parameters to fetch the timegraph tree.
-          Usually contains these parameters, but not limited to
-          - lowIndex, index of the first line to query
-          - size, number of lines to return
-          - columnIds, List of column ids to return, all columns will be returned if not specified. The data in the returned lines will use the same order as the columnId array
+        description: Query parameters to fetch the table lines.
+          One of 'requested_table_index' or 'requested_times' should be present. The 'requested_times' array should be of size 1.
```
Comment (bhufmann) [inline API.yaml:1222]: Briefly, explain what to use when

File: API.yaml [hunk 14]
```diff
@@ -1091,16 +1218,41 @@ paths:
...
+                parameters:
+                  type: object
+                  properties:
+                    requested_table_index:
+                      type: integer
+                    requested_times:
+                      type: array
+                      items:
+                        type: integer
+                    requested_table_count:
```
Comment (bhufmann) [inline API.yaml:1239]: not described in the description above. Please add

File: API.yaml [hunk 15]
```diff
@@ -1213,14 +1375,34 @@ paths:
           type: string
       requestBody:
         description: Query parameters to fetch the annotations.
-          Usually contains these parameters, but not limited to
-          - entries, array of all the entry (series) IDs
-          - times, array of times
+          When 'requested_times' size is 2 it is the start and end of the requested range, otherwise it is the explicit array of requested sample times.
```
Comment (bhufmann) [inline API.yaml:1378]: same comment as above

File: API.yaml [hunk 16]
```diff
@@ -1213,14 +1375,34 @@ paths:
...
+              type: object
+              properties:
+                parameters:
+                  type: object
+                  properties:
+                    requested_times:
+                      type: array
+                      items:
+                        type: integer
+                    requested_items:
```
Comment (bhufmann) [inline API.yaml:1392]: describe above in description

**Review by bhufmann — CHANGES_REQUESTED** [review]
It's getting much better and more understandable. Still some comments from my side.

**Review by bhufmann — APPROVED** [review]
Makes the TSP much more readable and understandable. Thanks for the updates.

**Review by MatthewKhouzam — APPROVED** [review]
I think this closes #46 too!

File: API.yaml (hunk 6 above)
Comment (michelrdagenais) [reply API.yaml:709]: The requested times could show more than 2 items in the example, to clearly show the difference (explicit array of requested sample times) with other places where the same name, requested times, is used for [start, end].

**Pull request merged**